### PR TITLE
Refactor to provide parity with future FusionAuth hosted token exchange

### PR DESCRIPTION
**Pull request by bhalsey**
## What is this PR and why do we need it?

FusionAuth will be adding token exchange endpoints.  This will eliminate the need to run your own server in addition to FusionAuth with a react app.  With this change, PKCE code generation is moved from the react app to the server side.   This change provides parity with the upcoming FusionAuth update and allows us to work with the refactored fusionauth-react-sdk.
https://github.com/FusionAuth/fusionauth-react-sdk/issues/34
https://github.com/FusionAuth/fusionauth-issues/issues/1991

### CODE REVIEW

File: kickstart/kickstart.json [hunk 1]
```diff
@@ -54,11 +54,12 @@
...
                 "name": "Example app",
                 "oauthConfiguration" : {
-                    "authorizedRedirectURLs": ["http://localhost:3000","http://localhost"],
-                    "authorizedOriginURLs": ["http://localhost:9000","http://localhost"],
+                    "authorizedRedirectURLs": ["http://localhost:3000","http://localhost:9000/app/token-exchange"],
+                    "authorizedOriginURLs": ["http://localhost:9000","http://localhost:3000"],
                     "clientSecret": "super-secret-secret-that-should-be-regenerated-for-production",
                     "logoutURL": "http://localhost:3000",
                     "enabledGrants": ["authorization_code","refresh_token"],
+                    "debug": true,
```
Comment (lyleschemmerling) [inline kickstart/kickstart.json:82]: We could additionally add the following to the kickstart to set up cors:

`{
          "method": "PATCH",
          "url": "api/system-configuration",
          "body":{
            "systemConfiguration": {
              "corsConfiguration": {
                "allowCredentials": true,
                "allowedMethods": [
                  "GET",
                  "POST",
                  "OPTIONS"
                ],
                "allowedOrigins": [
                  "http://localhost:3000"
                ],
                "debug": false,
                "enabled": true,
                "preflightMaxAgeInSeconds": 0
              }
            }
          }
        }`

File: README.md [hunk 2]
```diff
@@ -61,8 +61,8 @@ Go to the Applications section.
 
 1. Give it a name
 2. Make sure the authorization code grant is enabled
-3. Add the following to the authorized redirect URLs for your application: `http://localhost:3000` and `http://localhost`
-4. Add the following to the authorized request origins URLs for your application: `http://localhost:9000` and `http://localhost`
+3. Add the following to the authorized redirect URLs for your application: `http://localhost:3000` and `http://localhost` and localhost:9011 (??)
+4. Add the following to the authorized request origins URLs for your application: `http://localhost:9000` and `http://localhost` and localhost:9011 (??)
 5. Add the following to the logout URL: `http://localhost:3000`
 6. Click save.
 7. Edit the application again.
```
Comment (lyleschemmerling) [inline README.md]: for hosted we will want to go to system settings > CORS. Set enable cors true. Set allowed credentials true. Set allowed methods GET POST OPTIONS. Set allowed origins `http://localhost:3000`

**Review by lyleschemmerling — DISMISSED** [review]

File: kickstart/kickstart.json (hunk 1 above)
Comment (bhalsey) [reply kickstart/kickstart.json:82]: Nice.  done.

File: README.md (hunk 2 above)
Comment (bhalsey) [reply README.md]: Added those instructions in.  And this comment drew my attention to a needed fix.

**Review by lyleschemmerling — APPROVED** [review]

**Pull request merged**